.github/workflows/gcp-dsh-testing.yaml

```diff
@@ -0,0 +1,208 @@
+name: "[GCP] DevZero self-hosted deployment"
+run-name: '[GCP] self-hosted test run'
+
+on:
+  push:
+    branches:
+      - garvit/gcp-tf
+  workflow_dispatch:
+
+jobs:
+  setup-and-test:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: 'Authenticate to Google Cloud'
+        id: 'auth'
+        uses: 'google-github-actions/auth@v0.4.0'
+        with:
+          workload_identity_provider: 'projects/354385559261/locations/global/workloadIdentityPools/dz-ga-dsh-pool/providers/dz-ga-dsh-provider'
+          service_account: 'dsh-testing-github-actions@devzero-kubernetes-sandbox.iam.gserviceaccount.com'
+          create_credentials_file: true
+
+      - name: 'Set up Cloud SDK'
```

Comment:
This step is showing this warning `Warning: The gcloud CLI is not authenticated (or it is not installed). Authenticate by adding the "google-github-actions/auth" step prior this one.` likely because previous step is outdated